### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by nickstamas**
Hotttt

plasma-jest

### CODE REVIEW

**Comment by nickstamas** [comment]
My Saturday was lit 🔥

**Comment by yale** [comment]
@nickstamas wanna .gitignore the coverage files?

**Comment by yale** [comment]
sorry didn't mean to close >_<

**Comment by nickstamas** [comment]
@yale 🙏

File: src/components/Modal/Modal.jsx [hunk 1]
```diff
@@ -57,9 +57,7 @@ class Modal extends React.Component {
   }
 }
 
-Modal.defaultProps = {
-  minWidth: 'auto',
-};
+Modal.defaultProps = {};
```
Comment (aaroncouture) [inline src/components/Modal/Modal.jsx]: is this change going to break anything? If minWidth is null/undefined what will be passed to style?

File: src/components/Search/Search.jsx [hunk 2]
```diff
@@ -89,15 +90,19 @@ class Search extends Component {
...
+
 Search.propTypes = {
   clearable: PropTypes.bool,
   clearIconUrl: PropTypes.string,
   disabled: PropTypes.bool,
-  iconUrl: PropTypes.string.isRequired,
+  iconUrl: PropTypes.string,
   instructionText: PropTypes.string,
-  onChange: PropTypes.func.isRequired,
+  onChange: PropTypes.func,
```
Comment (aaroncouture) [inline src/components/Search/Search.jsx]: now that onChange is not required, we should do a check at L32

File: src/components/Search/Search.jsx [hunk 3]
```diff
@@ -89,15 +90,19 @@ class Search extends Component {
...
+Search.defaultProps = {
+  placeholder: 'Search...',
+};
+
 Search.propTypes = {
   clearable: PropTypes.bool,
   clearIconUrl: PropTypes.string,
   disabled: PropTypes.bool,
-  iconUrl: PropTypes.string.isRequired,
+  iconUrl: PropTypes.string,
```
Comment (aaroncouture) [inline src/components/Search/Search.jsx]: and what if iconUrl is empty?

File: src/components/Tag/Tag.jsx [hunk 4]
```diff
@@ -1,26 +1,38 @@
...
     >
-      <span>{text}</span>
+      <span>{text || children}</span>
     </div>
   );
 };
 
 Tag.propTypes = {
-  text: PropTypes.string.isRequired,
+  text: (props, propName, componentName) => {
```
Comment (aaroncouture) [inline src/components/Tag/Tag.jsx]: factor this out to textOrChildrenRequired

File: src/components/Modal/Modal.jsx (hunk 1 above)
Comment (wingyli) [inline src/components/Modal/Modal.jsx]: also since `minWidth` is not required

File: src/components/Tag/Tag.jsx (hunk 4 above)
Comment (aaroncouture) [inline src/components/Tag/Tag.jsx]:
```
// wayyy overkill
const requires = (predicate, message) => (props, propName, componentName) =>
  predicate(props)
    ? null
    : `${message} in ${componentName}`;

const textOrChildren = requires({text, children} => text || children, `'text' or 'children' prop required');

Tag.propTypes: {
text: textOrChildren,
children: textOrChildren,
}
```

File: src/tests/__tests__/Card.react-test.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    const component = shallow(
+      <Card
+        isExpanded
+        expandedComponent={expandedComponent}
+      >
+        YOLO
+      </Card>
+    );
+
+    expect(component.is('.isExpanded')).toBe(true);
```
Comment (yale) [inline src/tests/__tests__/Card.react-test.js]: is this matching on a class name, or prop?

Comment (nickstamas) [reply src/tests/__tests__/Card.react-test.js]: class name

**Comment by kangax** [comment]
@nickstamas hey, can we merge this in?

**Comment by nickstamas** [comment]
@kangax Yeah should be good. I don't have time to actually do the merge though. It's kinda old but shouldn't require too much effort.

File: src/components/Search/Search.jsx (hunk 3 above)
Comment (kangax) [reply src/components/Search/Search.jsx]: There'll be an empty image?

File: src/components/Search/Search.jsx (hunk 2 above)
Comment (kangax) [reply src/components/Search/Search.jsx]: fixed

File: src/components/Modal/Modal.jsx (hunk 1 above)
Comment (kangax) [reply src/components/Modal/Modal.jsx]: fixed

**Pull request merged**